src/gluonts/itertools.py

```diff
@@ -129,18 +149,79 @@ def __len__(self) -> int:
         return len(self.iterable)
 
 
-class IterableSlice(Iterable):
+# can't make this a dataclass because of pytorch-lightning assumptions
+class IterableSlice:
     """
     An iterable version of `itertools.islice`, i.e. one that can be iterated
-    over multiple times.
+    over multiple times:
+
+        >>> isl = IterableSlice([1, 2, 3, 4, 5], 3)
+        >>> list(isl)
+        [1, 2, 3]
+        >>> list(isl)
+        [4, 5]
+        >>> list(isl)
+        []
```

Comment:
I'm not sure this is correct, I would expect `isl` to materialize to the same list every time

Reply:
`IterableSlice(iter([1, 2, 3, 4, 5]), 3)` would give what the docstring says